javascript/dropzone/demo/UploadImage.php

```diff
@@ -0,0 +1,34 @@
+// This is just an example, change it to your specific needs
+// This example is intended for images
+
+<?php
+
+namespace Frontend\Modules\Blog\Ajax;
+
+use Frontend\Core\Engine\Base\AjaxAction;
+use Frontend\Core\Engine\Model as FrontendModel;
+
+class UploadImage extends AjaxAction
+{
+    public function execute()
+    {
+        $files = $_FILES['file'];
+        $return = array();
+        // Use a foreach because an upload can contain multiple files
+        foreach($files['name'] as $index => $file) {
+            $filename = md5($files['name'][$index] . time()) . '.jpeg';
+            if(move_uploaded_file($files['tmp_name'][$index], FRONTEND_FILES_PATH . '/Trips/Days/Images/source/' . $filename)) {
+                FrontendModel::generateThumbnails(
+                    FRONTEND_FILES_PATH . '/Blog/Days/Images/',
+                    FRONTEND_FILES_PATH . '/Blog/Days/Images/source/' . $filename
+                );
+            }
+            // Optional: delete source if you're handling big files
+            \SpoonFile::delete(FRONTEND_FILES_PATH . '/Trips/Days/Images/source/' . $filename);
+            $return[$index] = $filename;
+        }
+
+        // return the new filenames
+        $this->output(200, $return);
```

Comment:
Dit zou een iets properdere implementatie zijn. Het gebruikt geen global variables ($_FILES) en gebruikt de symfony methods om files te verplaatsen (moveFile)

Daarnaast is de extensie niet gehardcode. In jouw versie werd enkel jpeg ondersteund.

```php
        $files = array_map(
            function ($file) {
                $filename = md5(uniqid() . time()) . '.' . $file->getClientOriginalExtension();
                $file->move(
                    FRONTEND_FILES_PATH . '/Profiles/acquisitions/source',
                    $filename
                );
                Model::generateThumbnails(
                    FRONTEND_FILES_PATH . '/Profiles/acquisitions/',
                    FRONTEND_FILES_PATH . '/Profiles/acquisitions/source/' . $filename
                );

                return $filename;
            },
            $this->get('request')->files->all()
        );

        $this->output(self::OK, $files);
```